neuro_san/internals/run_context/langchain/token_counting/langchain_token_counter.py

```diff
@@ -97,50 +84,41 @@ async def count_tokens(self, awaitable: Awaitable) -> Any:
-        token_counter_context_manager = self.get_callback_for_llm(self.llm)
-
-        if token_counter_context_manager is not None:
-
-            # Record origin information in our own context var so we can associate
-            # with the langchain callback context vars more easily.
-            origin: List[Dict[str, Any]] = self.journal.get_origin()
-            origin_str: str = Origination.get_full_name_from_origin(origin)
-            ORIGIN_INFO.set(origin_str)
-
-            old_callback: Union[AsyncCallbackHandler, BaseCallbackHandler] = None
-            callback_var: ContextVar = self.get_context_var_for_llm(self.llm)
-            if callback_var is not None:
-                old_callback = callback_var.get()
-
-            # Use the context manager to count tokens as per
-            #   https://python.langchain.com/docs/how_to/llm_token_usage_tracking/#using-callbacks
-            #
-            # Caveats:
-            # * In using this context manager approach, any tool that is called
-            #   also has its token counts contributing to its callers for better or worse.
-            # * As of 2/21/25, it seems that tool-calling agents (branch nodes) are not
-            #   registering their tokens correctly. Not sure if this is a bug in langchain
-            #   or there is something we are not doing in that scenario that we should be.
-            with token_counter_context_manager() as callback:
-                # Create a new context for different ContextVar values
-                # and use the create_task() to run within that context.
-                new_context: Context = copy_context()
-                task: Task = new_context.run(self.create_task, awaitable)
-                retval = await task
-
-            if callback_var is not None:
-                callback_var.set(old_callback)
-        else:
-            # No token counting was available for the LLM, but we still need to invoke.
-            retval = await awaitable
+        callback: AsyncCallbackHandler = None
+        # Record origin information in our own context var so we can associate
+        # with the langchain callback context vars more easily.
+        origin: List[Dict[str, Any]] = self.journal.get_origin()
+        origin_str: str = Origination.get_full_name_from_origin(origin)
+        ORIGIN_INFO.set(origin_str)
+
+        # Use the context manager to count tokens as per
+        #   https://python.langchain.com/docs/how_to/llm_token_usage_tracking/#using-callbacks
+        #
+        # Caveats:
+        # * In using this context manager approach, any tool that is called
+        #   also has its token counts contributing to its callers for better or worse.
+        # * As of 2/21/25, it seems that tool-calling agents (branch nodes) are not
+        #   registering their tokens correctly. Not sure if this is a bug in langchain
+        #   or there is something we are not doing in that scenario that we should be.
```

Comment:
This 2/21/25 comment likely added by me is likely outdated and can be removed.
I believe this observation led to the original caveat of how we included subnetwork token counts.
If below comment needs to be removed, please get rid of this one too.